packages/valory/contracts/gnosis_safe_proxy_factory/contract.py

```diff
@@ -136,15 +136,18 @@ def build_tx_deploy_proxy_contract_with_nonce(  # pylint: disable=too-many-argum
         ):
             tx_parameters.update(ledger_api.try_get_gas_pricing())
 
-        if gas is not None:
-            tx_parameters["gas"] = Wei(gas)
+        tx_parameters["gas"] = (
+            Wei(gas) if gas != 0 else Wei(1)
+        )  # we set a value to avoid triggering the gas estimation during buildTransaction below
 
         if nonce is not None:
             tx_parameters["nonce"] = Nonce(nonce)
 
         transaction_dict = create_proxy_fn.buildTransaction(tx_parameters)
-        # Auto estimation of gas does not work. We use a little more gas just in case
-        transaction_dict["gas"] = Wei(transaction_dict["gas"] + 50000)
+        gas_estimate = ledger_api._try_get_gas_estimate(transaction_dict)
+        transaction_dict["gas"] = (
+            Wei(max(gas_estimate + 50000, gas)) if gas_estimate is not None else Wei(gas)
```

Comment:
Where is this 50K coming from, same for the 75K above?

Reply:
It's been here before. We took it from the safe repo. I didn't want to change it but mid term might want to.